README.md
# Flutter Auth Buttons

[![Build Status](https://travis-ci.org/dmjones/auth_social_buttons.svg?branch=master)](https://travis-ci.org/dmjones/auth_social_buttons)

Flutter widget library containing buttons for authenticating with popular social networks: Apple, Google, Facebook, Twitter
and Microsoft.

<img src="https://raw.githubusercontent.com/dmjones/auth_social_buttons/master/screenshots/example-app.png" alt="Screenshot" width="250" />

## Usage

Add `auth_social_buttons` to your `pubspec.yaml`, then import the Dart file:

```dart
import 'package:auth_social_buttons/auth_social_buttons.dart';
```

Use the `onPressed` attribute to capture the button press and call your authentication logic within that. To disable
the button, pass `null` or omit the attribute.

```dart
FacebookSignInButton(onPressed: () {
  // call authentication logic
});
```

Some buttons have a dark mode. Enable this with the optional parameter:

```dart
GoogleSignInButton(
  onPressed: () {/* ... */}, 
  darkMode: true, // default: false
)
```

You can adjust the border-radius of the buttons:

```dart
TwitterSignInButton(
  onPressed: () {},
  borderRadius: 10.0,
)
```

You can adjust the text style of the buttons:

```dart
TwitterSignInButton(
  onPressed: () {},
  textStyle: TextStyle(fontSize: 20, fontWeight: FontWeight.w700, fontFamily: "Roboto"),
)
```

You can adjust the splash color of the buttons:

```dart
GoogleSignInButton(
  onPressed: () {/* ... */}, 
  splashColor: Colors.white, 
  // setting splashColor to Colors.transparent will remove button ripple effect.
)
```

Buttons can be stretched like normal Material buttons. By default the button
contents are left-aligned when stretched. You can choose to center the icon and
text using the `centered` property.

```dart
TwitterSignInButton(
  onPressed: () {},
  centered: true,
)
```

See the documentation for API details: https://pub.dartlang.org/documentation/auth_social_buttons/latest/.

## Contributions

Contributions are very welcome. I would recommend discussing large changes in an issue before you spend the time on them.

Good quality pull requests will win you commit rights.
# auth_social_buttons
# auth_social_buttons


CHANGELOG.md
## [2.0.4]

* Change in Documentation.

## [2.0.3]

* Change in homepage.

## [2.0.2]

* Change in homepage.

## [2.0.1]

* Change in homepage.

## [2.0.0]

* Compatible for Flutter3.

## [0.10.0]

* Support for centered buttons (thanks [@Jwildsmith](https://github.com/Jwildsmith)).

## [0.9.0]

* Updated Facebook button (thanks [@chetan-cueclad](https://github.com/chetan-cueclad)).

## [0.8.0]

* Button splash color can be set (thanks [@bitmoxy](https://github.com/bitmoxy)).

## [0.7.0]

* Text styles can be overridden (thanks [@bitmoxy](https://github.com/bitmoxy)).

## [0.6.0]

* Added Apple button (thanks [@json469](https://github.com/json469)).
* Simplified the example app (removed stretched button examples).

## [0.5.0]

* Added Microsoft button.

## [0.4.0]

* Border radius can be specified (thanks [@nerder](https://github.com/nerder)).
* Google button shows correct background colour when pressed (thanks [@g3rrydanc3](https://github.com/g3rrydanc3)).
* Button content left-aligns when stretched.
* `onPressed` attribute is no longer required.

## [0.3.1]

* Improved package description.

## [0.3.0]

* Added Twitter button.

## [0.2.1]

* Fix bug to prevent buttons from stretching.

## [0.2.0]

* Added Facebook button.

## [0.1.0]

* Added example application.

## [0.0.1]

* Support for Google button.


test/apple_test.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/src/apple.dart';
import 'package:auth_social_buttons/src/button.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  testWidgets('Check the onTap of the Button works',
      (WidgetTester tester) async {
    var appleButtonKey = UniqueKey();
    var isTapped = false;

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(
            key: appleButtonKey,
            onPressed: () {
              isTapped = true;
            },
          ),
        ),
      ),
    );

    await tester.tap(find.byKey(appleButtonKey));
    expect(isTapped, true);
  });

  testWidgets('Check the default text provided is used',
      (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(),
        ),
      ),
    );

    expect(find.text('Sign in with Apple'), findsOneWidget);
  });

  testWidgets('Check the provided text is used', (WidgetTester tester) async {
    const buttonText = 'Sign up with Apple';
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(
            text: buttonText,
          ),
        ),
      ),
    );

    expect(find.text(buttonText), findsOneWidget);
  });

  testWidgets('Check the default text style of the button is used',
      (WidgetTester tester) async {
    const defaultTextStyle = TextStyle(
      fontSize: 16.0,
      fontWeight: FontWeight.w500,
      color: Colors.black,
    );

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(),
        ),
      ),
    );

    var text = find.byType(Text).evaluate().toList()[0].widget as Text;
    expect(text.style!.fontSize, defaultTextStyle.fontSize);
    expect(text.style!.fontWeight, defaultTextStyle.fontWeight);
    expect(text.style!.color, defaultTextStyle.color);
  });

  testWidgets('Check the provided text style of the button is used',
      (WidgetTester tester) async {
    const providedTextStyle = TextStyle(
      fontSize: 22.0,
      fontWeight: FontWeight.w900,
      color: Colors.white,
    );

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(
            textStyle: providedTextStyle,
          ),
        ),
      ),
    );

    final AppleSignInButton appleSignInButton =
        tester.firstWidget(find.byType(AppleSignInButton));
    expect(appleSignInButton.textStyle, providedTextStyle);
  });

  testWidgets('Check supplied splash color is used',
      (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(
            onPressed: () {},
            splashColor: Colors.white,
          ),
        ),
      ),
    );

    var button = find.byType(StretchableButton).evaluate().toList()[0].widget
        as StretchableButton;
    expect(button.splashColor, Colors.white);
  });

  testWidgets('Check default splash color is used',
      (WidgetTester tester) async {
    var buttonTheme = ButtonThemeData();

    await tester.pumpWidget(MaterialApp(
      home: Builder(
        builder: (BuildContext context) {
          buttonTheme = ButtonTheme.of(context);
          return AppleSignInButton(
            onPressed: () {},
          );
        },
      ),
    ));

    var button = find.byType(ElevatedButton).evaluate().toList()[0].widget
        as ElevatedButton;
    expect(buttonTheme.getSplashColor(button as MaterialButton),
        buttonTheme.getTextColor(button as MaterialButton).withOpacity(0.12));
  });

  testWidgets('Check the default AppleButtonStyle option is used',
      (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(),
        ),
      ),
    );
    final AppleSignInButton appleSignInButton =
        tester.firstWidget(find.byType(AppleSignInButton));
    expect(appleSignInButton.style, AppleButtonStyle.white);
  });

  testWidgets('Check the provided AppleButtonStyle option is used',
      (WidgetTester tester) async {
    final appleButtonStyle = AppleButtonStyle.black;

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(
            style: appleButtonStyle,
          ),
        ),
      ),
    );

    final AppleSignInButton appleSignInButton =
        tester.firstWidget(find.byType(AppleSignInButton));
    expect(appleSignInButton.style, appleButtonStyle);
  });

  testWidgets('Check the default border radius of the button is used',
      (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(),
        ),
      ),
    );
    final AppleSignInButton appleSignInButton =
        tester.firstWidget(find.byType(AppleSignInButton));
    expect(appleSignInButton.borderRadius, 3.0);
  });

  testWidgets('Check the provided border radius of the button is used',
      (WidgetTester tester) async {
    var appleButtonKey = UniqueKey();

    var borderRadius = 20.0;

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: AppleSignInButton(
            key: appleButtonKey,
            borderRadius: borderRadius,
          ),
        ),
      ),
    );

    final AppleSignInButton appleSignInButton =
        tester.firstWidget(find.byType(AppleSignInButton));
    expect(appleSignInButton.borderRadius, borderRadius);
  });
}


test/google_test.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/src/button.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:auth_social_buttons/auth_social_buttons.dart';

void main() {
  testWidgets('Check on tap works', (WidgetTester tester) async {
    var buttonKey = UniqueKey();

    var testVal = false;

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: GoogleSignInButton(
            key: buttonKey,
            onPressed: () {
              testVal = true;
            },
          ),
        ),
      ),
    );

    await tester.tap(find.byKey(buttonKey));
    expect(testVal, true);
  });

  testWidgets('Check default text is used', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: GoogleSignInButton(
            onPressed: () {},
          ),
        ),
      ),
    );

    expect(find.text("Sign in with Google"), findsOneWidget);
  });

  testWidgets('Check supplied text is used', (WidgetTester tester) async {
    const suppliedText = "Hello, World!";

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: GoogleSignInButton(
            onPressed: () {},
            text: suppliedText,
          ),
        ),
      ),
    );

    expect(find.text(suppliedText), findsOneWidget);
  });

  testWidgets('Check dark mode works', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: GoogleSignInButton(
            onPressed: () {},
            darkMode: true,
          ),
        ),
      ),
    );

    expect(find.byType(ElevatedButton), findsOneWidget);

    var button = find.byType(ElevatedButton).evaluate().toList()[0].widget
        as ElevatedButton;
    expect(button, Color(0xFF4285F4));

    var text = find.byType(Text).evaluate().toList()[0].widget as Text;
    expect(text.style!.color, Colors.white);
  });

  testWidgets('Check light mode works', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: GoogleSignInButton(
            onPressed: () {},
          ),
        ),
      ),
    );

    expect(find.byType(ElevatedButton), findsOneWidget);

    var button = find.byType(ElevatedButton).evaluate().toList()[0].widget
        as ElevatedButton;

    expect(button, Colors.white);

    var text = find.byType(Text).evaluate().toList()[0].widget as Text;
    expect(text.style!.color, Colors.black.withOpacity(0.54));
  });

  testWidgets('Check default text style is used', (WidgetTester tester) async {
    var defaultTextStyle = TextStyle(
        fontSize: 18.0,
        fontFamily: "Roboto",
        fontWeight: FontWeight.w500,
        color: Colors.black.withOpacity(0.54));

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: GoogleSignInButton(
            onPressed: () {},
          ),
        ),
      ),
    );

    var text = find.byType(Text).evaluate().toList()[0].widget as Text;
    expect(text.style!.fontSize, defaultTextStyle.fontSize);
    expect(text.style!.fontWeight, defaultTextStyle.fontWeight);
    expect(text.style!.color, defaultTextStyle.color);
  });

  testWidgets('Check supplied text style is used', (WidgetTester tester) async {
    const suppliedTextStyle = TextStyle(
        color: Color(0xff000000),
        fontSize: 20,
        fontWeight: FontWeight.w700,
        fontFamily: "Roboto");
    var buttonKey = UniqueKey();

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: GoogleSignInButton(
            key: buttonKey,
            onPressed: () {},
            textStyle: suppliedTextStyle,
          ),
        ),
      ),
    );

    final GoogleSignInButton button =
        tester.firstWidget(find.byType(GoogleSignInButton));
    expect(button.textStyle, suppliedTextStyle);
  });

  testWidgets('Check supplied splash color is used',
      (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: GoogleSignInButton(
            onPressed: () {},
            splashColor: Colors.white,
          ),
        ),
      ),
    );

    var button = find.byType(StretchableButton).evaluate().toList()[0].widget
        as StretchableButton;
    expect(button.splashColor, Colors.white);
  });

  testWidgets('Check default splash color is used',
      (WidgetTester tester) async {
    var buttonTheme = ButtonThemeData();

    await tester.pumpWidget(MaterialApp(
      home: Builder(
        builder: (BuildContext context) {
          buttonTheme = ButtonTheme.of(context);
          return GoogleSignInButton(
            onPressed: () {},
          );
        },
      ),
    ));

    var button = find.byType(ElevatedButton).evaluate().toList()[0].widget
        as ElevatedButton;
    expect(buttonTheme.getSplashColor(button as MaterialButton),
        buttonTheme.getTextColor(button as MaterialButton).withOpacity(0.12));
  });
}


test/twitter_test.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/src/button.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:auth_social_buttons/auth_social_buttons.dart';

void main() {
  testWidgets('Check on tap works', (WidgetTester tester) async {
    var buttonKey = UniqueKey();

    var testVal = false;

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: TwitterSignInButton(
            key: buttonKey,
            onPressed: () {
              testVal = true;
            },
          ),
        ),
      ),
    );

    await tester.tap(find.byKey(buttonKey));
    expect(testVal, true);
  });

  testWidgets('Check default text is used', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: TwitterSignInButton(
            onPressed: () {},
          ),
        ),
      ),
    );

    expect(find.text("Sign in with Twitter"), findsOneWidget);
  });

  testWidgets('Check supplied text is used', (WidgetTester tester) async {
    const suppliedText = "Hello, World!";

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: TwitterSignInButton(
            onPressed: () {},
            text: suppliedText,
          ),
        ),
      ),
    );

    expect(find.text(suppliedText), findsOneWidget);
  });

  testWidgets('Check default text style is used', (WidgetTester tester) async {
    const defaultTextStyle = TextStyle(
        fontSize: 18.0, fontWeight: FontWeight.bold, color: Color(0xFF555555));

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: TwitterSignInButton(
            onPressed: () {},
          ),
        ),
      ),
    );

    var text = find.byType(Text).evaluate().toList()[0].widget as Text;
    expect(text.style!.fontSize, defaultTextStyle.fontSize);
    expect(text.style!.fontWeight, defaultTextStyle.fontWeight);
    expect(text.style!.color, defaultTextStyle.color);
  });

  testWidgets('Check supplied text style is used', (WidgetTester tester) async {
    const suppliedTextStyle = TextStyle(
        color: Color(0xff000000),
        fontSize: 20,
        fontWeight: FontWeight.w700,
        fontFamily: "Roboto");
    var buttonKey = UniqueKey();

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: TwitterSignInButton(
            key: buttonKey,
            onPressed: () {},
            textStyle: suppliedTextStyle,
          ),
        ),
      ),
    );

    final TwitterSignInButton button =
        tester.firstWidget(find.byType(TwitterSignInButton));
    expect(button.textStyle, suppliedTextStyle);
  });

  testWidgets('Check supplied splash color is used',
      (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: TwitterSignInButton(
            onPressed: () {},
            splashColor: Colors.white,
          ),
        ),
      ),
    );

    var button = find.byType(StretchableButton).evaluate().toList()[0].widget
        as StretchableButton;
    expect(button.splashColor, Colors.white);
  });

  testWidgets('Check default splash color is used',
      (WidgetTester tester) async {
    var buttonTheme = ButtonThemeData();

    await tester.pumpWidget(MaterialApp(
      home: Builder(
        builder: (BuildContext context) {
          buttonTheme = ButtonTheme.of(context);
          return TwitterSignInButton(
            onPressed: () {},
          );
        },
      ),
    ));

    var button = find.byType(ElevatedButton).evaluate().toList()[0].widget
        as ElevatedButton;
    expect(buttonTheme.getSplashColor(button as MaterialButton),
        buttonTheme.getTextColor(button as MaterialButton).withOpacity(0.12));
  });
}


lib/src/apple.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/src/button.dart';

const double defaultBorderRadius = 3.0;

enum AppleButtonStyle { white, whiteOutline, black }

/// A sign in button that matches Apple's design guidelines.
class AppleSignInButton extends StatelessWidget {
  final String text;
  final AppleButtonStyle style;
  final double borderRadius;
  final VoidCallback? onPressed;
  final TextStyle? textStyle;
  final Color? splashColor;
  final bool centered;

  /// Creates a new button. Set [darkMode] to `true` to use the dark
  /// black background variant with white text, otherwise an all-white background
  /// with dark text is used.
  AppleSignInButton(
      {this.onPressed,
      // 'Continue with Apple' is also an available variant depdening on App's sign-in experience.
      this.text = 'Sign in with Apple',
      this.textStyle,
      this.splashColor,
      this.style = AppleButtonStyle.white,
      // Apple doesn't specify a border radius, but this looks about right.
      this.borderRadius = defaultBorderRadius,
      this.centered = false,
      Key? key})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return StretchableButton(
      buttonColor:
          style == AppleButtonStyle.black ? Colors.black : Colors.white,
      borderRadius: borderRadius,
      splashColor: splashColor,
      buttonBorderColor:
          style == AppleButtonStyle.whiteOutline ? Colors.black : null,
      onPressed: onPressed,
      buttonPadding: 0.0,
      centered: centered,
      children: <Widget>[
        Center(
          child: Row(
            children: <Widget>[
              Padding(
                padding: const EdgeInsets.only(left: 22.0, bottom: 3.0),
                child: Container(
                  height: 38.0,
                  width: 32.0,
                  decoration: BoxDecoration(
                    borderRadius: BorderRadius.circular(this.borderRadius),
                  ),
                  child: Center(
                    child: Image(
                      image: AssetImage(
                        "graphics/apple_logo_${style == AppleButtonStyle.black ? "white" : "black"}.png",
                        package: "auth_social_buttons",
                      ),
                      height: 17.0,
                      width: 17.0,
                    ),
                  ),
                ),
              ),
              Padding(
                padding: const EdgeInsets.fromLTRB(0.0, 8.0, 32.0, 8.0),
                child: Text(
                  text,
                  style: textStyle ??
                      TextStyle(
                        fontSize: 16.0,
                        fontFamily: "SF Pro",
                        fontWeight: FontWeight.w500,
                        color: style == AppleButtonStyle.black
                            ? Colors.white
                            : Colors.black,
                      ),
                ),
              ),
            ],
          ),
        )
      ],
    );
  }
}


lib/src/twitter.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/src/button.dart';

/// A sign in button that matches Twitter's look and feel.
///
/// The button text can be overridden, however the default text is recommended
/// in order to be compliant with the design guidelines and to maximise
/// conversion.
class TwitterSignInButton extends StatelessWidget {
  final String text;
  final TextStyle? textStyle;
  final VoidCallback? onPressed;
  final double borderRadius;
  final Color? splashColor;
  final bool centered;

  /// Creates a new button. The default button text is 'Sign in with Twitter'.
  TwitterSignInButton({
    this.onPressed,
    this.borderRadius = defaultBorderRadius,
    this.text = 'Sign in with Twitter',
    this.textStyle,
    this.splashColor,
    this.centered = false,
    Key? key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return StretchableButton(
      buttonColor: Color(0xFFE7E7E7),
      borderRadius: borderRadius,
      splashColor: splashColor,
      onPressed: onPressed,
      buttonBorderColor: Color(0xFFCCCCCC),
      buttonPadding: 0.0,
      centered: centered,
      children: <Widget>[
        // Facebook doesn't provide strict sizes, so this is a good
        // estimate of their examples within documentation.

        Image(
          image: AssetImage(
            "graphics/Twitter_Logo_Blue.png",
            package: "auth_social_buttons",
          ),
          height: 40.0,
          width: 40.0,
        ),
        Padding(
          padding: const EdgeInsets.only(right: 10.0),
          child: Text(
            text,
            style: textStyle ??
                TextStyle(
                  // default to the application font-style
                  fontSize: 18.0,
                  fontWeight: FontWeight.bold,
                  color: Color(0xFF555555),
                ),
          ),
        ),
      ],
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/auth_social_buttons.dart';

void main() async {
  runApp(new MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    const padding = 25.0;

    return MaterialApp(
      title: 'Button Demo',
      home: Scaffold(
        appBar: AppBar(
          title: Text("auth_social_buttons"),
        ),
        backgroundColor: Color.fromARGB(0xFF, 0xF0, 0xF0, 0xF0),
        body: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: <Widget>[
              Column(
                children: <Widget>[
                  SizedBox(height: padding),
                  AppleSignInButton(onPressed: () {}),
                  AppleSignInButton(
                      onPressed: () {}, style: AppleButtonStyle.whiteOutline),
                  AppleSignInButton(
                      onPressed: () {}, style: AppleButtonStyle.black),
                  SizedBox(height: padding),
                  GoogleSignInButton(onPressed: () {}),
                  GoogleSignInButton(onPressed: () {}, darkMode: true),
                  SizedBox(height: padding),
                  FacebookSignInButton(onPressed: () {}),
                  SizedBox(height: padding),
                  TwitterSignInButton(onPressed: () {}),
                  SizedBox(height: padding),
                  MicrosoftSignInButton(onPressed: () {}),
                  MicrosoftSignInButton(onPressed: () {}, darkMode: true),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}


lib/src/facebook.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/src/button.dart';

/// A sign in button that matches Facebook's design guidelines.
///
/// The button text can be overridden, however the default text is recommended
/// in order to be compliant with the design guidelines and to maximise
/// conversion.
class FacebookSignInButton extends StatelessWidget {
  final String text;
  final TextStyle? textStyle;
  final VoidCallback? onPressed;
  final double borderRadius;
  final Color? splashColor;
  final bool centered;

  /// Creates a new button. The default button text is 'Continue with Facebook',
  /// which apparently results in higher conversion. 'Login with Facebook' is
  /// another suggestion.
  FacebookSignInButton({
    this.onPressed,
    this.borderRadius = defaultBorderRadius,
    this.text = 'Continue with Facebook',
    this.textStyle,
    this.splashColor,
    this.centered = false,
    Key? key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return StretchableButton(
      buttonColor: Color(0xFF1877F2),
      borderRadius: borderRadius,
      splashColor: splashColor,
      onPressed: onPressed,
      buttonPadding: 8.0,
      centered: centered,
      children: <Widget>[
        // Facebook doesn't provide strict sizes, so this is a good
        // estimate of their examples within documentation.

        Image(
          image: AssetImage(
            "graphics/flogo-HexRBG-Wht-100.png",
            package: "auth_social_buttons",
          ),
          height: 24.0,
          width: 24.0,
        ),
        Padding(
          padding: const EdgeInsets.only(left: 6.0, right: 10.0),
          child: Text(
            text,
            style: textStyle ??
                TextStyle(
                  // default to the application font-style
                  fontSize: 18.0,
                  fontWeight: FontWeight.bold,
                  color: Colors.white,
                ),
          ),
        ),
      ],
    );
  }
}


lib/src/microsoft.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/src/button.dart';

/// A sign in button that matches Microsoft's design guidelines.
class MicrosoftSignInButton extends StatelessWidget {
  final String text;
  final TextStyle? textStyle;
  final VoidCallback? onPressed;
  final double borderRadius;
  final bool darkMode;
  final Color? splashColor;
  final bool centered;

  /// Creates a new button. The default button text is 'Sign in with Microsoft'.
  /// Microsoft also allows simply 'Sign in'.
  MicrosoftSignInButton({
    this.onPressed,
    this.borderRadius = 0.0,
    this.text = 'Sign in with Microsoft',
    this.textStyle,
    this.darkMode = false,
    this.splashColor,
    this.centered = false,
    Key? key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return StretchableButton(
      buttonColor: darkMode ? Color(0xFF2F2F2F) : Color(0xFFFFFFFF),
      borderRadius: borderRadius,
      splashColor: splashColor,
      buttonBorderColor: darkMode ? null : Color(0xFF8C8C8C),
      onPressed: onPressed,
      buttonPadding: 10.0, // This is an estimate
      centered: centered,
      children: <Widget>[
        Padding(
          padding: const EdgeInsets.only(left: 2.0), // adds to 10 to make 12
          child: Image(
            image: AssetImage(
              "graphics/ms-symbollockup_mssymbol_19.png",
              package: "auth_social_buttons",
            ),
            height: 21.0, // 41px - 2x 10px padding
            width: 21.0,
          ),
        ),
        Padding(
          padding: const EdgeInsets.only(left: 12.0, right: 2.0),
          child: Text(
            text,
            style: textStyle ??
                TextStyle(
                  // Should be Segoe 15px, but can't find that font and 15px
                  // seems too small...
                  fontSize: 18.0,
                  fontWeight: FontWeight.w600,
                  color: darkMode ? Colors.white : Color(0xFF5E5E5E),
                ),
          ),
        ),
      ],
    );
  }
}


lib/src/google.dart
import 'package:flutter/material.dart';
import 'package:auth_social_buttons/src/button.dart';

/// A sign in button that matches Google's design guidelines.
///
/// The button text can be overridden, however the default text is recommended
/// in order to be compliant with the design guidelines and to maximise
/// conversion.
class GoogleSignInButton extends StatelessWidget {
  final String text;
  final TextStyle? textStyle;
  final bool darkMode;
  final double borderRadius;
  final VoidCallback? onPressed;
  final Color? splashColor;
  final bool centered;

  /// Creates a new button. Set [darkMode] to `true` to use the dark
  /// blue background variant with white text, otherwise an all-white background
  /// with dark text is used.
  GoogleSignInButton(
      {this.onPressed,
      this.text = 'Sign in with Google',
      this.textStyle,
      this.splashColor,
      this.darkMode = false,
      // Google doesn't specify a border radius, but this looks about right.
      this.borderRadius = defaultBorderRadius,
      this.centered = false,
      Key? key})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return StretchableButton(
      buttonColor: darkMode ? Color(0xFF4285F4) : Colors.white,
      borderRadius: borderRadius,
      splashColor: splashColor,
      onPressed: onPressed,
      buttonPadding: 0.0,
      centered: centered,
      children: <Widget>[
        // The Google design guidelines aren't consistent. The dark mode
        // seems to have a perfect square of white around the logo, with a
        // thin 1dp (ish) border. However, since the height of the button
        // is 40dp and the logo is 18dp, it suggests the bottom and top
        // padding is (40 - 18) * 0.5 = 11. That's 10dp once we account for
        // the thin border.
        //
        // The design guidelines suggest 8dp padding to the left of the
        // logo, which doesn't allow us to center the image (given the 10dp
        // above). Something needs to give - either the 8dp is wrong or the
        // 40dp should be 36dp. I've opted to increase left padding to 10dp.
        Padding(
          padding: const EdgeInsets.all(1.0),
          child: Container(
            height: 38.0, // 40dp - 2*1dp border
            width: 38.0, // matches above
            decoration: BoxDecoration(
              color: darkMode ? Colors.white : null,
              borderRadius: BorderRadius.circular(this.borderRadius),
            ),
            child: Center(
              child: Image(
                image: AssetImage(
                  "graphics/google-logo.png",
                  package: "auth_social_buttons",
                ),
                height: 18.0,
                width: 18.0,
              ),
            ),
          ),
        ),

        SizedBox(width: 14.0 /* 24.0 - 10dp padding */),
        Padding(
          padding: const EdgeInsets.fromLTRB(0.0, 8.0, 8.0, 8.0),
          child: Text(
            text,
            style: textStyle ??
                TextStyle(
                  fontSize: 18.0,
                  fontFamily: "Roboto",
                  fontWeight: FontWeight.w500,
                  color:
                      darkMode ? Colors.white : Colors.black.withOpacity(0.54),
                ),
          ),
        ),
      ],
    );
  }
}
